plugins/mysql/mysqlembed.cpp

```diff
@@ -1080,14 +1156,67 @@ class MySQLEmbedFunctionContext : public CInterfaceOf<IEmbedFunctionContext>
                 cachedConnection = NULL;
             }
             conn.setown(new MySQLConnection(mysql_init(NULL)));
-            if (!mysql_real_connect(*conn, server, user, password, database, port, NULL, 0))
+            if (hasMySQLOpt)
             {
-                VStringBuffer err("mysql: failed to connect (%s)", mysql_error(*conn));
-                rtlFail(0, err.str());
+                ForEachItemIn(idx, opts)
+                {
+                    const char *opt = opts.item(idx);
+                    if (strnicmp(opt, "MYSQL_", 6)==0)
+                    {
+                        const char *val = strchr(opt, '=');
+                        StringBuffer optName(opt);
+                        if (val)
+                        {
+                            optName.setLength(val-opt);
+                            val++;
+                        }
+                        MySQLOptionDefinition &optDef = lookupOption(optName);
+                        int rc;
+                        if (optDef.paramType == ParamTypeNone)
+                        {
+                            if (val)
+                                failx("Option %s does not take a value", optName.str());
+                            rc = mysql_options(*conn, optDef.option, nullptr);
+                        }
+                        else
+                        {
+                            if (!val)
+                                failx("Option %s requires a value", optName.str());
+                            switch (optDef.paramType)
+                            {
+                            case ParamTypeString:
+                                rc = mysql_options(*conn, optDef.option, val);
+                                break;
+                            case ParamTypeUInt:
+                                {
+                                    unsigned int oval = strtoul(val, nullptr, 10);
```

Comment:
Sure, and for the most part it would take an egregious mistake by the developer. But a silly assumption such as MYSQL_OPT_CONNECT_TIMEOUT(.5) would be accepted, and rather than having a small timeout, it would result in timeout of 0.